osu.Game/Graphics/Containers/LogoTrackingContainer.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.MathUtils;
+using osu.Game.Screens.Menu;
+using osuTK;
+
+namespace osu.Game.Graphics.Containers
+{
+    /// <summary>
+    /// A container that handles tracking of an <see cref="OsuLogo"/> through different layout scenarios.
+    /// </summary>
+    public class LogoTrackingContainer : Container
+    {
+        public Facade LogoFacade { get; }
+
+        /// <summary>
+        /// Whether or not the logo assigned to this FacadeContainer should be tracking the position of its facade.
+        /// </summary>
+        public bool Tracking
+        {
+            get => tracking;
+            set
+            {
+                if (Logo != null)
+                {
+                    if (value && !tracking && Logo.IsTracking)
+                        throw new InvalidOperationException($"Cannot track an instance of {typeof(OsuLogo)} to multiple {typeof(LogoTrackingContainer)}s");
+
+                    Logo.IsTracking = value;
+                }
+
+                tracking = value;
+            }
+        }
+
+        protected OsuLogo Logo;
+
+        private float facadeScale;
+        private Easing easing;
+        private Vector2? startPosition;
+        private double? startTime;
+        private double duration;
+        private bool tracking;
+
+        public LogoTrackingContainer()
+        {
+            LogoFacade = new ExposedFacade();
+        }
+
+        /// <summary>
+        /// Assign the logo that should track the Facade's position, as well as how it should transform to its initial position.
+        /// </summary>
+        /// <param name="logo">The instance of the logo to be used for tracking.</param>
+        /// <param name="facadeScale">The scale of the facade. Does not actually affect the logo itself.</param>
+        /// <param name="duration">The duration of the initial transform. Default is instant.</param>
+        /// <param name="easing">The easing type of the initial transform.</param>
+        public void SetLogo(OsuLogo logo, float facadeScale = 1.0f, double duration = 0, Easing easing = Easing.None)
+        {
+            if (Logo != logo && Logo != null)
+            {
+                // If we're replacing the logo to be tracked, the old one no longer has a tracking container
+                Logo.IsTracking = false;
+            }
+
+            Logo = logo ?? throw new ArgumentNullException(nameof(logo));
```

Comment:
Argument validity checks should be the first thing that happens inside methods. Rewrite as:

```csharp
if (logo == null)
    throw ...

...

Logo = logo
```